lib/puppet_blacksmith/modulefile.rb

```diff
@@ -52,10 +52,16 @@ def version
       metadata['version']
     end
 
-    def bump!(level = :patch)
-      new_version = increase_version(version, level)
+    def bump_to_version!(new_version)
       text = File.read(path)
       text = replace_version(text, new_version)
+      File.open(path,"w") { |file| file.puts text }
+      new_version
+    end
+
+    def bump!(level = :patch)
+      new_version = increase_version(version, level)
+      bump_to_version!(new_version)
       File.open(path, "w") {|file| file.puts text}
```

Comment:
The ```bump_to_version!``` already writes the file and returns the new_version so you can remove this and the next line